website/docs/user_security.md

```diff
@@ -0,0 +1,237 @@
+        ValueRuleValidator('role', int),
+    ]
+
+    rules_validator = RuleValidator(allowed_update_types)
+
+    def __init__(self):
+        CRUD.__init__(self, User, database_handler, UserDataAccess.rules_validator)
+
+
+# USER SECURITY SETUP
+class PolicyRoles(enum.Enum):
+    USER = 1
+    TESTER = 2
+    ADMIN = 3
+
+
+class SessionUser(object):
+
+    def __init__(self, id: int, email: str):
+        self.id = id
+        self.email = email
+
+    def __dict__(self):
+        return {'id': self.id, 'email': self.email}
+
+
+class CustomerSecurity(UserSecurity):
+
+    def __init__(self, cookie_name: str, secret: str, expiration_time: timedelta):
+        UserSecurity.__init__(self, cookie_name, JWTTokenHandler(secret, expiration_time))
+
+    def secure_entry(self, request, session_obj: SessionUser, policies: list):
+        data_dict = result_to_dict(user_data_access.get(session_obj.id))
+        if data_dict['email'] == session_obj.email:
+            if data_dict['role'] in policies:
+                return response_status(HTTPStatus.OK)
+            else:
+                return response_status(HTTPStatus.UNAUTHORIZED)
+        else:
+            return response_status(HTTPStatus.FORBIDDEN)
+
+    def from_session_data(self, session_data: dict) -> SessionUser:
+        return SessionUser(session_data['id'], session_data['email'])
+
+    def generate_session_data(self, user) -> dict:
+        return {
+            'id': user['id'],
+            'email': user['email'],
+        }
+
+# CRUD INIT
+user_data_access = UserDataAccess()
+
+# SECURITY HANDLER REGISTER
+key = 'super-secret'
+security_handler = CustomerSecurity('user_cookie', key, timedelta(seconds=2))
+SecurityHandler.register(security_handler)
+
+
+# IMPLEMENTATION
+@RequireLogin(policies=[PolicyRoles.USER.value])
+@handle_exceptions
+def user_entry(request):
+    pass
+
+
+@RequireLogin(policies=[PolicyRoles.ADMIN.value])
+@handle_exceptions
+def admin_entry(request):
+    pass
+
+
+@RequireLogin(policies=[PolicyRoles.TESTER.value])
+@handle_exceptions
+def tester_entry(request):
+    pass
+
+
+@RequireLogin(policies=[PolicyRoles.TESTER.value, PolicyRoles.ADMIN.value])
+@handle_exceptions
+def tester_admin_entry(request):
+    pass
+
+# Function call with HttpRequest.COOKIES set as {'user_cookie': token}
+# where token is the JWT encoded token which will be decoded by UserSecurity
+# and the decoded object with data will be authenticated and authorized in secure_entry()
+response = user_entry(request)
+
+# Similarly, this process will happen with other function and the respective HTTP status code
+# will be returned. This is how UserSecurity is implemented in Core-lib
+
+# For authenticated user
+if response.status_code == 200:
+    user = result_to_dict(user_data_access.get(user_id))
+    # We can return this data back to our application for as an encoded JWT token 
+    user_session = SecurityHandler.get().generate_session_data_token(user)
+    
+

```

Comment:
@csrf_exempt
@require_POST
def api_login_facebook(request):
    ...
  body = request_body_dict(request)
email = body.get('email')
pass = body.get('pass')
   is_authenticated = auth_service.authnticate(email, pass)
  if is_authenticagted:
       user = ... get the user
       user_session = SecurityHandler.get().generate_session_data(user)
       response = response_json({'csrf_token': django.middleware.csrf.get_token(request), 'session': user_session})
       response.set_cookie(key=settings.COOKIE_NAME, value=SecurityHandler.get().generate_session_data_token(user))
      return response